Extensions/RosettaTraining/Scripts/TensorFlow/data_reader.py

```diff
@@ -0,0 +1,218 @@
+            minion['attack'],
+            self.from_bool(attackable),
+            self.from_bool(minion['taunt']),
+            self.from_bool(minion['divine_shield']),
+            self.from_bool(minion['stealth'])
+        ])
+
+    def read_minions_data(self, minions):
+        if minions is None:
+            minions = []
+
+        count = len(minions)
+
+        for idx in range(0, 7):
+            if idx < count:
+                self.add_minion_data(minions[idx])
+            else:
+                self.add_minion_data_placeholder()
+
+    def add_mana_crystal(self, mana_crystal):
+        self.data.extend([
+            mana_crystal['remaining'],
+            mana_crystal['total'],
+            mana_crystal['overload_owed']
+        ])
+
+    def add_current_hand(self, hand):
+        if hand is None:
+            hand = {}
+
+        playable = 0
+        for card in hand:
+            if card['playable']:
+                playable = playable + 1
+        self.data.append(playable)
+
+        def add_data(card_id, cost):
+            mapped_card_id = self.hand_card_mapper.get_index(card_id)
+            self.data.append(mapped_card_id)
+            self.data.append(cost)
+
+        for card in hand:
+            card_id = card['card_id']
+            cost = card['cost']
+            add_data(card_id, cost)
+        for _ in range(10 - len(hand)):
+            card_id = 0
+            cost = -1
+            add_data(card_id, cost)
+
+    def add_opponent_hand(self, hand):
+        if hand is None:
+            hand = {}
+        
+        self.data.extend([
+            len(hand)
+        ])
+
+    def add_hero_power(self, hero_power):
+        self.data.extend([
+            self.from_bool(hero_power['playable'])
+        ])
+
+    def read_field(self, board):
+        self.data.clear()
+
+        self.read_hero_data(board['current_player'])
+        self.read_hero_data(board['opponent_player'])
+        self.read_minions_data(board['current_player']['minions'])
+        self.read_minions_data(board['opponent_player']['minions'])
+        self.add_current_hand(board['current_player']['hand'])
+        self.add_opponent_hand(board['opponent_player']['hand'])
+
+        self.add_mana_crystal(board['current_player']['mana_crystal'])
+        self.add_hero_power(board['current_player']['hero_power'])
+
+        return list(self.data)
+
+    def get_label(self, current_player, player1_win):
+        player1_string = 'Player1'
+        player2_string = 'Player2'
+
+        if current_player == player1_string:
+            current_player_is_player1 = True
+        elif current_player == player2_string:
+            current_player_is_player1 = False
+        else:
+            assert False
+
+        assert player1_win is not None
+        if current_player_is_player1 is player1_win:
+            return LABEL_IF_PLAYER1_WIN
+        else:
+            return LABEL_IF_PLAYER1_LOSS
+
+    def win_or_loss(self, result):
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Method could be a function](https://app.codacy.com/manual/utilForever/RosettaStone/pullRequest?prid=4748929)